compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1907,6 +1923,19 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
                 String.valueOf(bLangSimpleVarRef.variableName);
         BSymbol resolvedSymbol = symResolver.lookupClosureVarSymbol(env,
                 Names.fromString(identifier), SymTag.VARIABLE);
+        
+        if (resolvedSymbol == symTable.notFoundSymbol 
+                && !withinLetClause && doClauseEnv != null 
+                && identifiers.containsKey(identifier) 
+                && withinQuery) {
+            resolvedSymbol = symResolver.lookupClosureVarSymbol(doClauseEnv, Names.fromString(identifier),
+                    SymTag.VARIABLE);
+            if (resolvedSymbol != symTable.notFoundSymbol) {
+                SymbolEnv queryLambdaEnv = SymbolEnv.createFuncBodyEnv(currentQueryLambdaBody, env);
```

Comment:
Why do we create a new env here? In the next line, we define the variable references within that scope. In my opinion, this is not correct. before visiting the BLangSimpleVarRef, symbols should be defined.

Reply:
We are creating the lambda body at the query desugar. While visiting this lambda body we have to create the symbols which we haven't defined previously.